[BUG] KudoSerializerTest crashed A HOST BUFFER WAS LEAKED intermittently

**Describe the bug**
first seen in jenkins-spark-rapids-jni_nightly-pre_release/501

mvn test failed clean up thread,
```
[2025-10-02T05:02:15.548Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project spark-rapids-jni: There are test failures.
[2025-10-02T05:02:15.548Z] [ERROR] 
[2025-10-02T05:02:15.548Z] [ERROR] Please refer to /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-pre_release-501-cuda13/target/surefire-reports for the individual test results.
[2025-10-02T05:02:15.548Z] [ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[2025-10-02T05:02:15.548Z] [ERROR] Error occurred in starting fork, check output in log
[2025-10-02T05:02:15.548Z] [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: Error occurred in starting fork, check output in log
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:624)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:954)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[2025-10-02T05:02:15.548Z] [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-02T05:02:15.548Z] [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2025-10-02T05:02:15.548Z] [ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-02T05:02:15.548Z] [ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
[2025-10-02T05:02:15.548Z] [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```


from com.nvidia.spark.rapids.jni.kudo.**KudoSerializerTest**-output.txt
```
...
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12353587 7f6bfe56c1a0)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12353586 7f6bfe56c180)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12353585 7f6bfe638990)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12353584 7f6bfe638970)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12353591 7f6bfe56c220)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12353590 7f6bfe56c200)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12353717 7f6bfe5f77f0)
[main] WARN com.nvidia.spark.rapids.jni.kudo.KudoSerializer - Dumped kudo tables to file: /tmp/kudo_dump_test2595583772575794802.bin
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12364727 7f6bfe69c180)
[Cleaner Thread] ERROR ai.rapids.cudf.ColumnVector - A DEVICE COLUMN VECTOR WAS LEAKED (ID: 12364714 7f6bfde7db70)
[Cleaner Thread] ERROR ai.rapids.cudf.ColumnVector - A DEVICE COLUMN VECTOR WAS LEAKED (ID: 12364708 7f6bfe670c80)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12364726 7f6bfd0f56e0)
[Cleaner Thread] ERROR ai.rapids.cudf.ColumnVector - A DEVICE COLUMN VECTOR WAS LEAKED (ID: 12364704 7f6bfe5aeed0)
[Cleaner Thread] ERROR ai.rapids.cudf.ColumnVector - A DEVICE COLUMN VECTOR WAS LEAKED (ID: 12364698 7f6bfe635e60)
[Cleaner Thread] ERROR ai.rapids.cudf.HostMemoryBuffer - A HOST BUFFER WAS LEAKED (ID: 12364725 7f6bfd0f56a0)
```

**Steps/Code to reproduce bug**
there is no simple repro

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/src/test/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializerTest.java
+++ b/src/test/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializerTest.java
@@ -627,7 +627,7 @@ public class KudoSerializerTest extends CudfTestBase {
   @Test
   public void testMergeWithDumpPath() {
     File tempFile = null;
-    try(CloseableArray<Table> tables = CloseableArray.wrap(new Table[4])) {
+    try(CloseableArray<Table> tables = CloseableArray.wrap(new Table[2])) {
       //Create a temporary file for dumping
       tempFile = File.createTempFile("kudo_dump_test", ".bin");
       tempFile.deleteOnExit();
@@ -674,7 +674,7 @@ public class KudoSerializerTest extends CudfTestBase {
           }
         };
         MergeOptions options = new MergeOptions(DumpOption.Always, outputStreamSupplier, dumpPath);
-        serializer.mergeOnHost(kudoTables.getArray(), options);
+        serializer.mergeOnHost(kudoTables.getArray(), options).close();
 
         // Verify dump file exists and has content
         assertTrue(tempFile.exists(), "Dump file should exist");

